src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,94 +53,1321 @@
+                                                    teamMember, bundle.getFullNameFromRoster(teamMember),
+                                                    new ArrayList<InstructorResultsQuestionTable>(), getProfilePictureLink(teamMember), 
+                                                    viewType.isFirstGroupedByGiver(), moderationButton);
+
+            giverPanel.setHasResponses(false);
+            addParticipantPanelToSectionPanel(sectionPanel, teamName, giverPanel);
+        }
+    }
+    
+    private void addMissingParticipantsForTeamToSectionPanelWithoutModerationButton(
+                                    InstructorFeedbackResultsSectionPanel sectionPanel, 
+                                    String teamName, List<String> teamMembers) {
+        for (String teamMember : teamMembers) {
+            InstructorFeedbackResultsGroupByQuestionPanel giverPanel = 
+                InstructorFeedbackResultsGroupByQuestionPanel.buildInstructorFeedbackResultsGroupByQuestionPanelWithoutModerationButton(
+                    new ArrayList<InstructorResultsQuestionTable>(), getProfilePictureLink(teamMember), 
+                    viewType.isFirstGroupedByGiver(), teamMember, bundle.getFullNameFromRoster(teamMember));
+                
+            giverPanel.setHasResponses(false);
+            
+            addParticipantPanelToSectionPanel(sectionPanel, teamName, giverPanel);
+        }
+    }
+    
+
+    private void addParticipantPanelToSectionPanel(
+                                    InstructorFeedbackResultsSectionPanel sectionPanel, String currentTeam,
+                                    InstructorFeedbackResultsGroupByQuestionPanel giverPanel) {
+        List<InstructorResultsParticipantPanel> teamsMembersPanels;
+        
+        if (sectionPanel.getParticipantPanels().containsKey(currentTeam)) {
+            teamsMembersPanels = sectionPanel.getParticipantPanels().get(currentTeam);
+        } else {
+            teamsMembersPanels = new ArrayList<InstructorResultsParticipantPanel>();
+        }
+        
+        teamsMembersPanels.add(giverPanel);
+        sectionPanel.getParticipantPanels().put(currentTeam, teamsMembersPanels);
+    }
+
+    /**
+     * 
+     * 
+     * @param sectionPanel
+     * @param questions
+     * @param responsesGroupedByTeam
+     * @param teamsInSection
+     */
+    private void buildTeamsStatisticsTableForSectionPanel(
+                     InstructorFeedbackResultsSectionPanel sectionPanel, 
+                     LinkedHashMap<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> responsesGroupedByTeam,
+                     Collection<String> teamsInSection) {
+        Map<String, List<InstructorResultsQuestionTable>> teamToStatisticsTables = new HashMap<String, List<InstructorResultsQuestionTable>>();
+        for (String team : teamsInSection) {
```

Comment:
Would it be better to be more specific and go with `Set<String>` for `teamsInSection`?
(Like in the `LinkedHashMap` case.)